microprofile-lra/README.adoc

```diff
@@ -0,0 +1,592 @@
+        "that was assigned %s participant Id.", lraId, participantId);
+
+    return failLRA ? Response.status(Response.Status.INTERNAL_SERVER_ERROR).build() : Response.ok().build();
+}
+----
+
+In this `GET` JAX-RS method, we also use the `@LRA` annotation that either starts a new LRA or joins an existing one
+which is defined by the default LRA type `REQUIRED`. This is decided based on the `LRA.LRA_HTTP_CONTEXT_HEADER` header
+we called `lraId`. If the framework starts a new LRA,
+this header is automatically populated with its ID. If the caller specifies this `LRA.LRA_HTTP_CONTEXT_HEADER`
+manually in the request, the received LRA is joined. As you can see, the LRA context or ID is propagated in HTTP
+headers.
+
+The second header parameter `LRA.LRA_HTTP_RECOVERY_HEADER` is considered a unique participant ID for a particular
+enlistment within LRA. If we would like to enlist `LRAParticipant1` in the same LRA (`LRA.LRA_HTTP_CONTEXT_HEADER`)
+multiple times, this recovery ID would be different so we can associate the invocations of compensate and complete
+methods.
+
+Each LRA participant needs to define the `@Compensate` method that defines the compensating action.
+
+[source,java]
+----
+@Compensate
+@PUT
+@Path("/compensate")
+public Response compensateWork(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                               @HeaderParam(LRA.LRA_HTTP_RECOVERY_HEADER) URI participantId) {
+    LOGGER.infof("Compensating action for Participant 1 (%s) in LRA %s.", participantId, lraId);
+    return Response.ok().build();
+}
+----
+
+The compensation is defined by the `@Compensate` annotation which needs to be placed on the JAX-RS PUT method so the LRA
+coordinator knows how to call it. For simplicity, we are just printing the messages to the console. The participant can
+control how it finishes its participation in LRA via the returned status code. Please see the
+link:https://github.com/eclipse/microprofile-lra/blob/main/spec/src/main/asciidoc/microprofile-lra-spec.asciidoc[specification]
+for more details.
+
+The complete method looks similarly. It uses the `@Complete` annotation and it also needs to be the JAX-RS PUT method.
+
+[source,java]
+----
+@Complete
+@PUT
+@Path("/complete")
+public Response completeWork(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                             @HeaderParam(LRA.LRA_HTTP_RECOVERY_HEADER) URI participantId) {
+    LOGGER.infof("Complete action for Participant 1 (%s) in LRA %s.", participantId, lraId);
+    return Response.ok().build();
+}
+----
+
+The LRA coordinator invokes the `@Compensate` method when the LRA cancels on failure and it invokes the `@Complete`
+method when the LRA closes successfully.
+
+NOTE: The `@Complete` and `@Compensate` methods don't need to be JAX-RS methods. See the specification for details.
+
+Now we are already able to start our first LRA. You can deploy the application to the {productName} as demonstrated in
```

Comment:
Unfortunately, we are not able. There is missing ParticipantResult.java code in the Readme